### PR TITLE
[ResourceDetectors.Azure] Update AzureVmMetaDataRequestor [Add timeout for HttpClient]

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #.

## Changes

Adding a timeout to reduce start up time when the application is not running on Azure VM.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ internal static class AzureVmMetaDataRequestor
 
     public static AzureVmMetadataResponse? GetAzureVmMetaDataResponseDefault()
     {
-        using var httpClient = new HttpClient();
+        using var httpClient = new HttpClient() { Timeout = TimeSpan.FromSeconds(2) };
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs:31]: We might need to consider a 1 sec, it's good to start with 2 secs delay.

Comment (utpilla) [reply src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs:31]: Why are we changing the default value?

Comment (vishweshbankwar) [reply src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs:31]: The default timeout 100 secs. For applications running in non-Azure VM environments this results in longer start up time for app.

Comment (utpilla) [reply src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs:31]: Would this throw an exception when timed out?

Comment (vishweshbankwar) [reply src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs:31]: Yes.

Comment (utpilla) [reply src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs:31]: Okay just checked that it would be caught in `Detect`.

**Review by utpilla — APPROVED** [review]

**Pull request merged**